### PR TITLE
Added condition for '+' case.

### LINKED ISSUE #13018: Unary `+` parameter triggers overeager breaking of method chain
**Prettier 2.7.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzCEAUB9AhgSgDoAjXAJ0yIBs44ZKBuAHSnSwFoCTzLCa7GLNpgDUXUhWq16+BiAA0ICAAcYAS2gBnZKHJkIAdwAK5BNpS4qB3AE9ti4mVxgA1nQDKuALZwAMmqg4ZFRLTTgHJ1cPZWcAgHNkGDIAV3CQOC9iOAATbJzfXCg45Nw4uAAxCDIvXBh1IuQQXGSYCAUQAAsYLyoAdQ61eE0YsDh3M0G1ADdBm0awTXsQALCyGCMnOJrg0LSAK00AD3d4mgBFZIh4HaowxRiyVcbSLKp25TIAmF61bJgO5AADgADPd9GFek5lI0PnBVlMgooAI6XeAbFTmJqadiBHI5dpkOAotSEjalbZIEK3NJhLxqRIpGmnOAXK5BSm7RQwXDEH5-AFIABMXKcaio8QAwhAvBT0poAKztZJhAAqPPMVLuICmqQAklA8rB3GBPqoAIIG9wwGw0G5hAC+9qAA)

```sh
--parser babel
```

**Input:**

```jsx
foo(_a).bar().leet();
foo(-a).bar().leet();
foo(+a).bar().leet();
```

**Output:**

```jsx
foo(_a).bar().leet();
foo(-a).bar().leet();
foo(+a)
  .bar()
  .leet();

```

**Expected behavior:**
```jsx
foo(_a).bar().leet();
foo(-a).bar().leet();
foo(+a).bar().leet();
```

### PR DESCRIPTION
**Pull request by t-mangoe**
## Description

this PR fixes the issue #13018 

Fix #13018

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by t-mangoe** [comment]
I am not sure how to update the test, so, if the test needs to be updated, I would appreciate knowing how to update it.
Also, if there are any other areas that need correction, please let me know.

**Comment by sosukesuzuki** [comment]
@t-mangoe Thank you for working on this! I've pushed a commit to add tests. If you will contribute to Prettier again, please add tests with to refer my commit.

**Review by sosukesuzuki — APPROVED** [review]

**Comment by fisker** [comment]
should we also do the same to  [`~`](https://github.com/estree/estree/blob/master/es5.md#unaryoperator) and [`++`](https://github.com/estree/estree/blob/master/es5.md#updateoperator)?

**Comment by sosukesuzuki** [comment]
And `--`

**Comment by fisker** [comment]
> `node.operator.length <= 2` ?

No, just add the `UpdateExpression` and list operators.

**Comment by t-mangoe** [comment]
@fisker
Thank you for your comment. I added processing for `UpdateExpression` case. Also, I grouped the target operators into `targetUnaryExpressionOperators` and `targetUpdateExpressionOperators`.
With these responses, how do you like it?

File: tests/format/js/method-chain/13018.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
 foo(_a).bar().leet();
 foo(-a).bar().leet();
 foo(+a).bar().leet();
+foo(~a).bar().leet();
+foo(++a).bar().leet();
```
Comment (fisker) [inline tests/format/js/method-chain/13018.js:5]: I assume we are going to treat `a++`  the same, please add test for it.

Comment (t-mangoe) [reply tests/format/js/method-chain/13018.js:5]: I added tests for `a++` and `a--`.

**Review by fisker — APPROVED** [review]
Looks good to me, thank you!

**Comment by fisker** [comment]
> Also, I grouped the target operators into targetUnaryExpressionOperators and targetUpdateExpressionOperators.

Sorry, missed this.

I'd use

```js
const simpleOperators = new Map([
	'UnaryExpression', new Set(['!', '-', '+', '~']),
	// To prevent unexpected operators add in future
	'UpdateExpression', new Set(['++', '--']),
]);

if (simpleOperators.has(node.type) && simpleOperators.get(node.type).has(node.operator)) {
  // ...
}

```

**Pull request merged**